File: src/main/java/com/regionrental/commands/RemoveCommand.java
package com.regionrental.commands;

import com.regionrental.RegionRental;
import com.regionrental.managers.Rental;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

import java.util.Map;

/**
 * Command to completely remove RegionRental setup from a region
 * This removes signs, schematics, and configuration data
 * Useful when a region needs to be repurposed or is no longer needed for rentals
 */
public class RemoveCommand implements CommandExecutor {

    private final RegionRental plugin;

    public RemoveCommand(RegionRental plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (!sender.hasPermission("regionrental.admin.remove")) {
            sender.sendMessage(plugin.getConfigManager().getMessage("no-permission"));
            return true;
        }

        if (args.length < 1) {
            sender.sendMessage(ChatColor.RED + "Usage: /rrremove <region>");
            sender.sendMessage(ChatColor.YELLOW + "This will completely remove RegionRental setup from the region.");
            sender.sendMessage(ChatColor.YELLOW + "If the region is currently rented, the rental will be reset with full refund.");
            return true;
        }

        String regionName = args[0];

        // Check if WorldGuard region exists
        if (!plugin.getWorldGuardManager().regionExists(regionName)) {
            sender.sendMessage(plugin.getConfigManager().getMessage("region-not-found",
                "{region}", regionName));
            return true;
        }

        // Check if region has an active rental
        Rental rental = plugin.getRentalManager().getRental(regionName);
        if (rental != null) {
            // Reset the rental with net refund first (prevents double-refunds)
            Map<String, Object> refundDetails = plugin.getRentalManager().resetRentalWithRefund(regionName);

            if (refundDetails != null) {
                String playerName = (String) refundDetails.get("playerName");
                double refundAmount = (double) refundDetails.get("refundAmount");
                double totalPaid = (double) refundDetails.get("totalPaid");
                double alreadyRefunded = (double) refundDetails.get("alreadyRefunded");

                String formattedAmount = String.format(plugin.getConfigManager().getCurrencyFormat(), refundAmount);
                String formattedTotal = String.format(plugin.getConfigManager().getCurrencyFormat(), totalPaid);
                String formattedAlready = String.format(plugin.getConfigManager().getCurrencyFormat(), alreadyRefunded);

                sender.sendMessage(ChatColor.YELLOW + "Active rental found. Player " + playerName +
                        " has been refunded " + formattedAmount);

                // Show refund breakdown if any previous refunds exist
                if (alreadyRefunded > 0) {
                    sender.sendMessage(ChatColor.GRAY + "  Total paid: " + ChatColor.YELLOW + formattedTotal);
                    sender.sendMessage(ChatColor.GRAY + "  Already refunded: " + ChatColor.YELLOW + formattedAlready);
                    sender.sendMessage(ChatColor.GRAY + "  Net refund: " + ChatColor.GREEN + formattedAmount);
                }
            }
        }

        // Remove rental sign
        boolean signRemoved = plugin.getSignManager().removeRegionSetup(regionName);

        // Delete WorldEdit schematic if it exists
        boolean schematicDeleted = false;
        if (plugin.getWorldEditManager().hasCapture(regionName)) {
            plugin.getWorldEditManager().deleteCapture(regionName);
            schematicDeleted = true;
        }

        // Send comprehensive success message
        StringBuilder message = new StringBuilder();
        message.append(plugin.getConfigManager().getMessage("region-removed",
            "{region}", regionName));

        if (signRemoved) {
            message.append("\n").append(ChatColor.GREEN).append("  ✓ Rental sign removed");
        }

        if (schematicDeleted) {
            message.append("\n").append(ChatColor.GREEN).append("  ✓ WorldEdit schematic deleted");
        }

        if (rental != null) {
            message.append("\n").append(ChatColor.GREEN).append("  ✓ Active rental reset with refund");
        }

        sender.sendMessage(message.toString());

        // Log the action
        plugin.getLogger().info("Admin " + sender.getName() + " removed RegionRental setup from region: " + regionName);

        return true;
    }
}

File: src/main/kotlin/com/zonerental/commands/RemoveCommand.kt
package com.zonerental.commands

import com.zonerental.ZoneRental
import com.zonerental.extensions.sendMiniMessage
import com.zonerental.util.WorldRegionParser
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

/**
 * Command to completely remove ZoneRental setup from a region
 * This removes signs, schematics, and configuration data
 * Useful when a region needs to be repurposed or is no longer needed for rentals
 */
class RemoveCommand(private val plugin: ZoneRental) : CommandExecutor {

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (!sender.hasPermission("zonerental.admin.remove")) {
            sender.sendMessage(plugin.configManager.getMessage("no-permission"))
            return true
        }

        if (args.isEmpty()) {
            sender.sendMiniMessage("<red>Usage: /zrremove <world:region>")
            sender.sendMiniMessage("<yellow>Example: /zrremove world:shop1")
            sender.sendMiniMessage("<yellow>This will completely remove ZoneRental setup from the region.")
            sender.sendMiniMessage("<yellow>If the region is currently rented, the rental will be reset with full refund.")
            return true
        }

        // Parse region argument with world inference
        val parsed = WorldRegionParser.parse(args[0], sender) ?: run {
            sender.sendMiniMessage("<red>Invalid format! Console must use world:region format (e.g., world:shop1)")
            return true
        }

        val world = parsed.getWorld() ?: run {
            sender.sendMiniMessage("<red>World not found!")
            return true
        }
        val regionName = parsed.regionName

        // Check if WorldGuard region exists
        if (!plugin.worldGuardManager.regionExists(regionName, world)) {
            sender.sendMessage(plugin.configManager.getMessage("region-not-found",
                "{region}", parsed.getCompositeKey()))
            return true
        }

        // Check if region has an active rental
        val rental = plugin.rentalManager.getRental(regionName, world)
        if (rental != null) {
            // Reset the rental with net refund first (prevents double-refunds)
            val refundDetails = plugin.rentalManager.resetRentalWithRefund(regionName, world)

            if (refundDetails != null) {
                val playerName = refundDetails["playerName"] as String
                val refundAmount = refundDetails["refundAmount"] as Double
                val totalPaid = refundDetails["totalPaid"] as Double
                val alreadyRefunded = refundDetails["alreadyRefunded"] as Double

                val formattedAmount = String.format(plugin.configManager.currencyFormat, refundAmount)
                val formattedTotal = String.format(plugin.configManager.currencyFormat, totalPaid)
                val formattedAlready = String.format(plugin.configManager.currencyFormat, alreadyRefunded)

                sender.sendMiniMessage("<yellow>Active rental found. Player $playerName has been refunded $formattedAmount")

                // Show refund breakdown if any previous refunds exist
                if (alreadyRefunded > 0) {
                    sender.sendMiniMessage("<gray>  Total paid: <yellow>$formattedTotal")
                    sender.sendMiniMessage("<gray>  Already refunded: <yellow>$formattedAlready")
                    sender.sendMiniMessage("<gray>  Net refund: <green>$formattedAmount")
                }
            }
        }

        // Remove rental sign
        val signRemoved = plugin.signManager.removeRegionSetup(regionName, world)

        // Delete WorldEdit schematic if it exists
        val schematicDeleted = if (plugin.worldEditManager.hasCapture(regionName, world)) {
            plugin.worldEditManager.deleteCapture(regionName, world)
            true
        } else {
            false
        }

        // Check if region is in a group and remove it
        val compositeKey = parsed.getCompositeKey()
        val groupName = plugin.groupsConfig.getRegionGroup(compositeKey)
        val groupRemoved = if (groupName != null) {
            plugin.groupsConfig.removeRegionsFromGroup(groupName, listOf(compositeKey))
        } else {
            false
        }

        // Only remove individual overrides if region was NOT in a group
        // (group membership already clears individual overrides when joining)
        val regionConfigRemoved = if (!groupRemoved && plugin.regionsConfig.hasRegion(regionName, world)) {
            plugin.regionsConfig.removeRegion(regionName, world)
            true
        } else {
            false
        }

        // Send comprehensive success message
        val message = buildString {
            append(plugin.configManager.getMessage("region-removed", "{region}", parsed.getCompositeKey()))

            if (signRemoved) {
                append("\n<green>  ✓ Rental sign removed")
            }

            if (schematicDeleted) {
                append("\n<green>  ✓ WorldEdit schematic deleted")
            }

            if (regionConfigRemoved) {
                append("\n<green>  ✓ Region configuration removed from regions.yml")
            }

            if (groupRemoved) {
                append("\n<green>  ✓ Removed from group '$groupName'")
            }

            if (rental != null) {
                append("\n<green>  ✓ Active rental reset with refund")
            }
        }

        sender.sendMiniMessage(message)

        // Log the action
        plugin.logger.info("Admin ${sender.name} removed ZoneRental setup from region: ${parsed.getCompositeKey()}")

        return true
    }
}
